test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -196,34 +185,31 @@ class database_operation {
              */
             tc->sleep();
 
-            /* Pick a random collection to update if there are any, taking care to subtract -1. */
-            collection_count = tc->database.get_collection_count();
-            if (collection_count == 0)
-                continue;
-
-            collection_id =
-              random_generator::instance().generate_integer<uint64_t>(0, collection_count - 1);
-
-            if (collections.find(collection_id) == collections.end()) {
-                WT_CURSOR *random_cursor = nullptr, *update_cursor = nullptr;
-                /* Retrieve the collection name associated with our id. */
-                collection_name = std::move(tc->database.get_collection_name(collection_id));
-                debug_print("Thread {" + std::to_string(tc->id) +
-                    "} Creating cursor for collection: " + collection_name,
-                  DEBUG_TRACE);
-
-                /* Open a random cursor for that collection. */
-                tc->session->open_cursor(tc->session, collection_name.c_str(), nullptr,
-                  "next_random=true", &random_cursor);
-                /*
-                 * We can't call update on a random cursor so we open two cursors here, one to do
-                 * the randomized next and one to subsequently update the key.
-                 */
-                tc->session->open_cursor(
-                  tc->session, collection_name.c_str(), nullptr, nullptr, &update_cursor);
-
-                collections.emplace(
-                  collection_id, collection_cursors{collection_name, random_cursor, update_cursor});
+            /* Pick a random collection to update. */
+            {
```

Comment:
Is there a specific reason why you need a new block here ?

Reply:
Is it for the shared_ptr to destruct itself after each iteration ?
Would be it be unsafe to remove those brackets though ?